static/third/bootstrap-typeahead/typeahead.js

```diff
@@ -141,7 +183,10 @@
         // converts the input text to html elements.
         this.$element.html('');
       } else {
-        this.$element.val(this.updater(val, e)).trigger("change");
+        const [to_replace, replacement, diff_index] = get_string_diff(this.$element.val(), this.updater(val, e))
```

Comment:
It looks like `to_replace` isn't used directly -- just its length; does it make sense to have `get_string_diff` just return the length?

Reply:
Indeed! It'd be even better if the function returned:

```
const [replacement, from, to] = get_replacement_info(a,b);
```

that can then be directly fed into the editing API.